File: db-4.8.30/csharp/ReplicationStats.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;

namespace BerkeleyDB {
    /// <summary>
    /// Statistical information about the replication subsystem
    /// </summary>
    public class ReplicationStats {
        private Internal.ReplicationStatStruct st;
        private LSN next;
        private LSN waiting;
        private LSN maxPerm;
        private LSN winner;
        internal ReplicationStats(Internal.ReplicationStatStruct stats) {
            st = stats;
            next = new LSN(st.st_next_lsn.file, st.st_next_lsn.offset);
            waiting = new LSN(st.st_waiting_lsn.file, st.st_waiting_lsn.offset);
            maxPerm =
                new LSN(st.st_max_perm_lsn.file, st.st_max_perm_lsn.offset);
            winner = new LSN(st.st_election_lsn.file, st.st_election_lsn.offset);

        }

        /// <summary>
        /// Log records currently queued. 
        /// </summary>
        public ulong CurrentQueuedLogRecords { get { return st.st_log_queued; } }
        /// <summary>
        /// Site completed client sync-up. 
        /// </summary>
        public bool ClientStartupComplete { get { return st.st_startup_complete != 0; } }
        /// <summary>
        /// Current replication status. 
        /// </summary>
        public uint Status { get { return st.st_status; } }
        /// <summary>
        /// Next LSN to use or expect. 
        /// </summary>
        public LSN NextLSN { get { return next; } }
        /// <summary>
        /// LSN we're awaiting, if any. 
        /// </summary>
        public LSN AwaitedLSN { get { return waiting; } }
        /// <summary>
        /// Maximum permanent LSN. 
        /// </summary>
        public LSN MaxPermanentLSN { get { return maxPerm; } }
        /// <summary>
        /// Next pg we expect. 
        /// </summary>
        public uint NextPage { get { return st.st_next_pg; } }
        /// <summary>
        /// pg we're awaiting, if any. 
        /// </summary>
        public uint AwaitedPage { get { return st.st_waiting_pg; } }
        /// <summary>
        /// # of times a duplicate master condition was detected.
        /// </summary>
        public uint DupMasters { get { return st.st_dupmasters; } }
        /// <summary>
        /// Current environment ID. 
        /// </summary>
        public int EnvID { get { return st.st_env_id; } }
        /// <summary>
        /// Current environment priority. 
        /// </summary>
        public uint EnvPriority { get { return st.st_env_priority; } }
        /// <summary>
        /// Bulk buffer fills. 
        /// </summary>
        public ulong BulkBufferFills { get { return st.st_bulk_fills; } }
        /// <summary>
        /// Bulk buffer overflows. 
        /// </summary>
        public ulong BulkBufferOverflows { get { return st.st_bulk_overflows; } }
        /// <summary>
        /// Bulk records stored. 
        /// </summary>
        public ulong BulkRecordsStored { get { return st.st_bulk_records; } }
        /// <summary>
        /// Transfers of bulk buffers. 
        /// </summary>
        public ulong BulkBufferTransfers { get { return st.st_bulk_transfers; } }
        /// <summary>
        /// Number of forced rerequests. 
        /// </summary>
        public ulong ForcedRerequests { get { return st.st_client_rerequests; } }
        /// <summary>
        /// Number of client service requests received by this client.
        /// </summary>
        public ulong ClientServiceRequests { get { return st.st_client_svc_req; } }
        /// <summary>
        /// Number of client service requests missing on this client.
        /// </summary>
        public ulong ClientServiceRequestsMissing { get { return st.st_client_svc_miss; } }
        /// <summary>
        /// Current generation number. 
        /// </summary>
        public uint CurrentGenerationNumber { get { return st.st_gen; } }
        /// <summary>
        /// Current election gen number. 
        /// </summary>
        public uint CurrentElectionGenerationNumber { get { return st.st_egen; } }
        /// <summary>
        /// Log records received multiply. 
        /// </summary>
        public ulong DuplicateLogRecords { get { return st.st_log_duplicated; } }
        /// <summary>
        /// Max. log records queued at once. 
        /// </summary>
        public ulong MaxQueuedLogRecords { get { return st.st_log_queued_max; } }
        /// <summary>
        /// Total # of log recs. ever queued. 
        /// </summary>
        public ulong QueuedLogRecords { get { return st.st_log_queued_total; } }
        /// <summary>
        /// Log records received and put. 
        /// </summary>
        public ulong ReceivedLogRecords { get { return st.st_log_records; } }
        /// <summary>
        /// Log recs. missed and requested. 
        /// </summary>
        public ulong MissedLogRecords { get { return st.st_log_requested; } }
        /// <summary>
        /// Env. ID of the current master. 
        /// </summary>
        public int MasterEnvID { get { return st.st_master; } }
        /// <summary>
        /// # of times we've switched masters. 
        /// </summary>
        public ulong MasterChanges { get { return st.st_master_changes; } }
        /// <summary>
        /// Messages with a bad generation #. 
        /// </summary>
        public ulong BadGenerationMessages { get { return st.st_msgs_badgen; } }
        /// <summary>
        /// Messages received and processed. 
        /// </summary>
        public ulong ReceivedMessages { get { return st.st_msgs_processed; } }
        /// <summary>
        /// Messages ignored because this site was a client in recovery.
        /// </summary>
        public ulong IgnoredMessages { get { return st.st_msgs_recover; } }
        /// <summary>
        /// # of failed message sends. 
        /// </summary>
        public ulong FailedMessageSends { get { return st.st_msgs_send_failures; } }
        /// <summary>
        /// # of successful message sends. 
        /// </summary>
        public ulong MessagesSent { get { return st.st_msgs_sent; } }
        /// <summary>
        /// # of NEWSITE msgs. received. 
        /// </summary>
        public ulong NewSiteMessages { get { return st.st_newsites; } }
        /// <summary>
        /// Current number of sites we will assume during elections.
        /// </summary>        
        public uint Sites { get { return st.st_nsites; } }
        /// <summary>
        /// # of times we were throttled. 
        /// </summary>
        public ulong Throttled { get { return st.st_nthrottles; } }
        /// <summary>
        /// # of times we detected and returned an OUTDATED condition.
        /// </summary>
        public ulong Outdated { get { return st.st_outdated; } }
        /// <summary>
        /// Pages received multiply. 
        /// </summary>
        public ulong DuplicatePages { get { return st.st_pg_duplicated; } }
        /// <summary>
        /// Pages received and stored. 
        /// </summary>
        public ulong ReceivedPages { get { return st.st_pg_records; } }
        /// <summary>
        /// Pages missed and requested. 
        /// </summary>
        public ulong MissedPages { get { return st.st_pg_requested; } }
        /// <summary>
        /// # of transactions applied. 
        /// </summary>
        public ulong AppliedTransactions { get { return st.st_txns_applied; } }
        /// <summary>
        /// # of STARTSYNC msgs delayed. 
        /// </summary>
        public ulong StartSyncMessagesDelayed { get { return st.st_startsync_delayed; } }

        /* Elections generally. */
        /// <summary>
        /// # of elections held. 
        /// </summary>
        public ulong Elections { get { return st.st_elections; } }
        /// <summary>
        /// # of elections won by this site. 
        /// </summary>
        public ulong ElectionsWon { get { return st.st_elections_won; } }

        /* Statistics about an in-progress election. */
        /// <summary>
        /// Current front-runner. 
        /// </summary>
        public int CurrentWinner { get { return st.st_election_cur_winner; } }
        /// <summary>
        /// Election generation number. 
        /// </summary>
        public uint ElectionGenerationNumber { get { return st.st_election_gen; } }
        /// <summary>
        /// Max. LSN of current winner. 
        /// </summary>
        public LSN CurrentWinnerMaxLSN { get { return winner; } }
        /// <summary>
        /// # of "registered voters". 
        /// </summary>
        public uint RegisteredSites { get { return st.st_election_nsites; } }
        /// <summary>
        /// # of "registered voters" needed. 
        /// </summary>
        public uint RegisteredSitesNeeded { get { return st.st_election_nvotes; } }
        /// <summary>
        /// Current election priority. 
        /// </summary>
        public uint ElectionPriority { get { return st.st_election_priority; } }
        /// <summary>
        /// Current election status. 
        /// </summary>
        public int ElectionStatus { get { return st.st_election_status; } }
        /// <summary>
        /// Election tiebreaker value. 
        /// </summary>
        public uint ElectionTiebreaker { get { return st.st_election_tiebreaker; } }
        /// <summary>
        /// Votes received in this round. 
        /// </summary>
        public uint Votes { get { return st.st_election_votes; } }
        /// <summary>
        /// Last election time seconds. 
        /// </summary>
        public uint ElectionTimeSec { get { return st.st_election_sec; } }
        /// <summary>
        /// Last election time useconds. 
        /// </summary>
        public uint ElectionTimeUSec { get { return st.st_election_usec; } }
        /// <summary>
        /// Maximum lease timestamp seconds. 
        /// </summary>
        public uint MaxLeaseSec { get { return st.st_max_lease_sec; } }
        /// <summary>
        /// Maximum lease timestamp useconds. 
        /// </summary>
        public uint MaxLeaseUSec { get { return st.st_max_lease_usec; } }
    }
}

File: db-4.8.30/java/src/com/sleepycat/db/ReplicationStats.java
/*-
 * Automatically built by dist/s_java_stat.
 * Only the javadoc comments can be edited.
 *
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2002-2009 Oracle.  All rights reserved.
 */

package com.sleepycat.db;

import com.sleepycat.db.internal.DbConstants;

/**
Replication statistics for a database environment.
*/
public class ReplicationStats {
    // no public constructor
    /* package */ ReplicationStats() {}
    /**
    The environment is configured as a replication client, as reported by {@link #getStatus}.
    */
    public static final int REP_CLIENT = DbConstants.DB_REP_CLIENT;

    /**
    The environment is configured as a replication master, as reported by {@link #getStatus}.
    */
    public static final int REP_MASTER = DbConstants.DB_REP_MASTER;

    /**
    Replication is not configured for this environment, as reported by {@link #getStatus}.
    */
    public static final int REP_NONE = 0;

    private long st_log_queued;
    /** TODO */
    public long getLogQueued() {
        return st_log_queued;
    }

    private int st_startup_complete;
    /** TODO */
    public boolean getStartupComplete() {
        return (st_startup_complete != 0);
    }

    private int st_status;
    /** TODO */
    public int getStatus() {
        return st_status;
    }

    private LogSequenceNumber st_next_lsn;
    /** TODO */
    public LogSequenceNumber getNextLsn() {
        return st_next_lsn;
    }

    private LogSequenceNumber st_waiting_lsn;
    /** TODO */
    public LogSequenceNumber getWaitingLsn() {
        return st_waiting_lsn;
    }

    private LogSequenceNumber st_max_perm_lsn;
    /** TODO */
    public LogSequenceNumber getMaxPermLsn() {
        return st_max_perm_lsn;
    }

    private int st_next_pg;
    /** TODO */
    public int getNextPages() {
        return st_next_pg;
    }

    private int st_waiting_pg;
    /** TODO */
    public int getWaitingPages() {
        return st_waiting_pg;
    }

    private int st_dupmasters;
    /** TODO */
    public int getDupmasters() {
        return st_dupmasters;
    }

    private int st_env_id;
    /** TODO */
    public int getEnvId() {
        return st_env_id;
    }

    private int st_env_priority;
    /** TODO */
    public int getEnvPriority() {
        return st_env_priority;
    }

    private long st_bulk_fills;
    /** TODO */
    public long getBulkFills() {
        return st_bulk_fills;
    }

    private long st_bulk_overflows;
    /** TODO */
    public long getBulkOverflows() {
        return st_bulk_overflows;
    }

    private long st_bulk_records;
    /** TODO */
    public long getBulkRecords() {
        return st_bulk_records;
    }

    private long st_bulk_transfers;
    /** TODO */
    public long getBulkTransfers() {
        return st_bulk_transfers;
    }

    private long st_client_rerequests;
    /** TODO */
    public long getClientRerequests() {
        return st_client_rerequests;
    }

    private long st_client_svc_req;
    /** TODO */
    public long getClientSvcReq() {
        return st_client_svc_req;
    }

    private long st_client_svc_miss;
    /** TODO */
    public long getClientSvcMiss() {
        return st_client_svc_miss;
    }

    private int st_gen;
    /** TODO */
    public int getGen() {
        return st_gen;
    }

    private int st_egen;
    /** TODO */
    public int getEgen() {
        return st_egen;
    }

    private long st_log_duplicated;
    /** TODO */
    public long getLogDuplicated() {
        return st_log_duplicated;
    }

    private long st_log_queued_max;
    /** TODO */
    public long getLogQueuedMax() {
        return st_log_queued_max;
    }

    private long st_log_queued_total;
    /** TODO */
    public long getLogQueuedTotal() {
        return st_log_queued_total;
    }

    private long st_log_records;
    /** TODO */
    public long getLogRecords() {
        return st_log_records;
    }

    private long st_log_requested;
    /** TODO */
    public long getLogRequested() {
        return st_log_requested;
    }

    private int st_master;
    /** TODO */
    public int getMaster() {
        return st_master;
    }

    private long st_master_changes;
    /** TODO */
    public long getMasterChanges() {
        return st_master_changes;
    }

    private long st_msgs_badgen;
    /** TODO */
    public long getMsgsBadgen() {
        return st_msgs_badgen;
    }

    private long st_msgs_processed;
    /** TODO */
    public long getMsgsProcessed() {
        return st_msgs_processed;
    }

    private long st_msgs_recover;
    /** TODO */
    public long getMsgsRecover() {
        return st_msgs_recover;
    }

    private long st_msgs_send_failures;
    /** TODO */
    public long getMsgsSendFailures() {
        return st_msgs_send_failures;
    }

    private long st_msgs_sent;
    /** TODO */
    public long getMsgsSent() {
        return st_msgs_sent;
    }

    private long st_newsites;
    /** TODO */
    public long getNewsites() {
        return st_newsites;
    }

    private int st_nsites;
    /** TODO */
    public int getNumSites() {
        return st_nsites;
    }

    private long st_nthrottles;
    /** TODO */
    public long getNumThrottles() {
        return st_nthrottles;
    }

    private long st_outdated;
    /** TODO */
    public long getOutdated() {
        return st_outdated;
    }

    private long st_pg_duplicated;
    /** TODO */
    public long getPagesDuplicated() {
        return st_pg_duplicated;
    }

    private long st_pg_records;
    /** TODO */
    public long getPagesRecords() {
        return st_pg_records;
    }

    private long st_pg_requested;
    /** TODO */
    public long getPagesRequested() {
        return st_pg_requested;
    }

    private long st_txns_applied;
    /** TODO */
    public long getTxnsApplied() {
        return st_txns_applied;
    }

    private long st_startsync_delayed;
    /** TODO */
    public long getStartSyncDelayed() {
        return st_startsync_delayed;
    }

    private long st_elections;
    /** TODO */
    public long getElections() {
        return st_elections;
    }

    private long st_elections_won;
    /** TODO */
    public long getElectionsWon() {
        return st_elections_won;
    }

    private int st_election_cur_winner;
    /** TODO */
    public int getElectionCurWinner() {
        return st_election_cur_winner;
    }

    private int st_election_gen;
    /** TODO */
    public int getElectionGen() {
        return st_election_gen;
    }

    private LogSequenceNumber st_election_lsn;
    /** TODO */
    public LogSequenceNumber getElectionLsn() {
        return st_election_lsn;
    }

    private int st_election_nsites;
    /** TODO */
    public int getElectionNumSites() {
        return st_election_nsites;
    }

    private int st_election_nvotes;
    /** TODO */
    public int getElectionNumVotes() {
        return st_election_nvotes;
    }

    private int st_election_priority;
    /** TODO */
    public int getElectionPriority() {
        return st_election_priority;
    }

    private int st_election_status;
    /** TODO */
    public int getElectionStatus() {
        return st_election_status;
    }

    private int st_election_tiebreaker;
    /** TODO */
    public int getElectionTiebreaker() {
        return st_election_tiebreaker;
    }

    private int st_election_votes;
    /** TODO */
    public int getElectionVotes() {
        return st_election_votes;
    }

    private int st_election_sec;
    /** TODO */
    public int getElectionSec() {
        return st_election_sec;
    }

    private int st_election_usec;
    /** TODO */
    public int getElectionUsec() {
        return st_election_usec;
    }

    private int st_max_lease_sec;
    /** TODO */
    public int getMaxLeaseSec() {
        return st_max_lease_sec;
    }

    private int st_max_lease_usec;
    /** TODO */
    public int getMaxLeaseUsec() {
        return st_max_lease_usec;
    }

    /**
    For convenience, the ReplicationStats class has a toString method
    that lists all the data fields.
    */
    public String toString() {
        return "ReplicationStats:"
            + "\n  st_log_queued=" + st_log_queued
            + "\n  st_startup_complete=" + (st_startup_complete != 0)
            + "\n  st_status=" + st_status
            + "\n  st_next_lsn=" + st_next_lsn
            + "\n  st_waiting_lsn=" + st_waiting_lsn
            + "\n  st_max_perm_lsn=" + st_max_perm_lsn
            + "\n  st_next_pg=" + st_next_pg
            + "\n  st_waiting_pg=" + st_waiting_pg
            + "\n  st_dupmasters=" + st_dupmasters
            + "\n  st_env_id=" + st_env_id
            + "\n  st_env_priority=" + st_env_priority
            + "\n  st_bulk_fills=" + st_bulk_fills
            + "\n  st_bulk_overflows=" + st_bulk_overflows
            + "\n  st_bulk_records=" + st_bulk_records
            + "\n  st_bulk_transfers=" + st_bulk_transfers
            + "\n  st_client_rerequests=" + st_client_rerequests
            + "\n  st_client_svc_req=" + st_client_svc_req
            + "\n  st_client_svc_miss=" + st_client_svc_miss
            + "\n  st_gen=" + st_gen
            + "\n  st_egen=" + st_egen
            + "\n  st_log_duplicated=" + st_log_duplicated
            + "\n  st_log_queued_max=" + st_log_queued_max
            + "\n  st_log_queued_total=" + st_log_queued_total
            + "\n  st_log_records=" + st_log_records
            + "\n  st_log_requested=" + st_log_requested
            + "\n  st_master=" + st_master
            + "\n  st_master_changes=" + st_master_changes
            + "\n  st_msgs_badgen=" + st_msgs_badgen
            + "\n  st_msgs_processed=" + st_msgs_processed
            + "\n  st_msgs_recover=" + st_msgs_recover
            + "\n  st_msgs_send_failures=" + st_msgs_send_failures
            + "\n  st_msgs_sent=" + st_msgs_sent
            + "\n  st_newsites=" + st_newsites
            + "\n  st_nsites=" + st_nsites
            + "\n  st_nthrottles=" + st_nthrottles
            + "\n  st_outdated=" + st_outdated
            + "\n  st_pg_duplicated=" + st_pg_duplicated
            + "\n  st_pg_records=" + st_pg_records
            + "\n  st_pg_requested=" + st_pg_requested
            + "\n  st_txns_applied=" + st_txns_applied
            + "\n  st_startsync_delayed=" + st_startsync_delayed
            + "\n  st_elections=" + st_elections
            + "\n  st_elections_won=" + st_elections_won
            + "\n  st_election_cur_winner=" + st_election_cur_winner
            + "\n  st_election_gen=" + st_election_gen
            + "\n  st_election_lsn=" + st_election_lsn
            + "\n  st_election_nsites=" + st_election_nsites
            + "\n  st_election_nvotes=" + st_election_nvotes
            + "\n  st_election_priority=" + st_election_priority
            + "\n  st_election_status=" + st_election_status
            + "\n  st_election_tiebreaker=" + st_election_tiebreaker
            + "\n  st_election_votes=" + st_election_votes
            + "\n  st_election_sec=" + st_election_sec
            + "\n  st_election_usec=" + st_election_usec
            + "\n  st_max_lease_sec=" + st_max_lease_sec
            + "\n  st_max_lease_usec=" + st_max_lease_usec
            ;
    }
}
